common/libs/wallet/non-deterministic/helpers.ts

```diff
@@ -33,22 +33,33 @@ export const signWrapper = (walletToWrap: IFullWallet): WrappedWallet =>
   });
 
 function determineKeystoreType(file: string): string {
-  const parsed = JSON.parse(file);
-  if (parsed.encseed) {
-    return KeystoreTypes.presale;
-  } else if (parsed.Crypto || parsed.crypto) {
-    return KeystoreTypes.utc;
-  } else if (parsed.hash && parsed.locked === true) {
-    return KeystoreTypes.v1Encrypted;
-  } else if (parsed.hash && parsed.locked === false) {
-    return KeystoreTypes.v1Unencrypted;
-  } else if (parsed.publisher === 'MyEtherWallet') {
-    return KeystoreTypes.v2Unencrypted;
-  } else {
-    throw new Error('Invalid keystore');
+  try {
+    const parsed = JSON.parse(file);
+    if (parsed.encseed) {
+      return KeystoreTypes.presale;
+    } else if (parsed.Crypto || parsed.crypto) {
+      return KeystoreTypes.utc;
+    } else if (parsed.hash && parsed.locked === true) {
+      return KeystoreTypes.v1Encrypted;
+    } else if (parsed.hash && parsed.locked === false) {
+      return KeystoreTypes.v1Unencrypted;
+    } else if (parsed.publisher === 'MyEtherWallet') {
+      return KeystoreTypes.v2Unencrypted;
+    } else {
+      return '';
+    }
+  } catch {
+    return '';
```

Comment:
Do we really want to return an empty string? What's the situation in which that's the desired return value?

Reply:
The function is intended to return a string. Returning an empty string is falsey and cuts down on the amount of `try catch` statements if I just threw an error in the final `else {}` statement

Reply:
@wbobeirne Maybe you have some thoughts on how we could make this more intuitive (or confirm you agree this is the best approach).